packages/excubiae/test/EASExcubia.test.ts

```diff
@@ -0,0 +1,130 @@
+import { expect } from "chai"
+import { ethers } from "hardhat"
+import { BigNumberish, Signer, ZeroAddress, getDefaultProvider } from "ethers"
+import { EASExcubia, EASExcubia__factory, MockEAS, MockEAS__factory } from "../typechain-types"
+
+describe("EASExcubia", function () {
+    let MockEASContract: MockEAS__factory
+    let EASExcubiaContract: EASExcubia__factory
+    let easExcubia: EASExcubia
+
+    let signer: Signer
+    let signerAddress: string
+
+    let gate: Signer
+    let gateAddress: string
+
+    let mockEAS: MockEAS
+    let mockEASAddress: string
+
+    const schemaId = "0xfdcfdad2dbe7489e0ce56b260348b7f14e8365a8a325aef9834818c00d46b31b"
+    const validAttestationId = ethers.encodeBytes32String("0x01")
+    const revokedAttestationId = ethers.encodeBytes32String("0x02")
+    const invalidSchemaAttestationId = ethers.encodeBytes32String("0x03")
+    const invalidRecipientAttestationId = ethers.encodeBytes32String("0x04")
+    const invalidAttesterAttestationId = ethers.encodeBytes32String("0x05")
+
+    before(async function () {
+        ;[signer, gate] = await ethers.getSigners()
+        signerAddress = await signer.getAddress()
+        gateAddress = await gate.getAddress()
+
+        MockEASContract = await ethers.getContractFactory("MockEAS")
+        mockEAS = await MockEASContract.deploy(signerAddress, signerAddress, schemaId)
+        mockEASAddress = await mockEAS.getAddress()
+
+        EASExcubiaContract = await ethers.getContractFactory("EASExcubia")
+        easExcubia = await EASExcubiaContract.deploy(mockEASAddress, signerAddress, schemaId)
+    })
+
+    describe("Deployment", function () {
+        it("Should deploy the EASExcubia contract correctly", async function () {
+            expect(easExcubia).to.not.eq(undefined)
+        })
+
+        it("Should deploy the MockEAS contract correctly", async function () {
+            expect(mockEAS).to.not.eq(undefined)
+        })
+
+        it("Should fail to deploy EASExcubia when attester parameter is not valid", async () => {
+            await expect(EASExcubiaContract.deploy(signerAddress, ZeroAddress, schemaId)).to.be.revertedWithCustomError(
+                easExcubia,
+                "ZeroAddress"
+            )
+        })
+    })
+
+    describe("EASExcubia", function () {
+        it("should fail to set the gate when the caller is not the owner", async () => {
+            const [, notOwnerSigner] = await ethers.getSigners()
+
+            await expect(easExcubia.connect(notOwnerSigner).setGate(gateAddress)).to.be.revertedWithCustomError(
+                easExcubia,
+                "OwnableUnauthorizedAccount"
+            )
+        })
+
+        it("should fail to set the gate when the gate address is zero", async () => {
```

Comment:
I see we are testing the abstract contract functions, so maybe worth testing that you cannot re-set it? https://github.com/privacy-scaling-explorations/zk-kit.solidity/blob/20e2c671665e8f18de6ed6b488339040d16a72d1/packages/excubiae/contracts/Excubia.sol#L27

Reply:
+1